helix-term/src/args.rs

```diff
@@ -97,6 +97,14 @@ impl Args {
             args.files.push(parse_file(&arg));
         }
 
+        // Set args.working_directory to the first file if it is a directory + replace it with "."
+        if let Some((path, _)) = args.files.first_mut() {
```

Comment:
Should we also check that `args.working_directory.is_none()` here? I think that flag should take precedence over the first file/dir

Reply:
In the original code flow, the directory in the first file took precedence. This is why I kept it that way.

We could change the old behavior, and leave files[0] untouched when the working dir is set, and I think the behaviour would make sense logically (like `tar -C <dir>`)

Reply:
Sorry, if the path was relative before, I guess it would work as if working dir took precedence. I just thought about it wrong.

Will fix

Reply:
I think the way to fix this is to store the directory from `args.files[0]` in a new member such as `args.extra_working_directory`, and then leave the existing logic. Then, in `main()` set the working directory twice, in order.

I will make this change later today unless I'm told otherwise.

Reply:
My last suggestion also has an edge case that doesn't work, will need to consider it more